### DISCUSSION TITLE
Transparent mode shows nothing captured

### DISCUSSION THREAD

**Discussion by BeyondEvil** [mitmproxy/mitmproxy#4985] [Q&A]
On my iPhone 12 mini running iOS 14.8.1 I have a 3rd party application from which I want to capture all the traffic.

I first tried mitmproxy in "regular" mode and was able to capture some traffic, but clearly not all.

So I wanted to try to see if I could capture all the traffic using "transparent" mode.

I've followed the instructions [here](https://docs.mitmproxy.org/stable/howto-transparent/#macos)

```sh
$ echo 'rdr pass on en0 inet proto tcp to any port {80, 443} -> 127.0.0.1 port 8080' > pf.conf
$ cat pf.conf
rdr pass on en0 inet proto tcp to any port {80, 443} -> 127.0.0.1 port 8080
```

```sh
$ sudo pfctl -f pf.conf 
pfctl: Use of -f option, could result in flushing of rules
present in the main ruleset added by the system at startup.
See /etc/pf.conf for further details.

No ALTQ support in kernel
ALTQ related functions disabled
```

```sh
$ sudo pfctl -e
No ALTQ support in kernel
ALTQ related functions disabled
pfctl: pf already enabled
```

```sh
$ sudo cat /etc/sudoers | grep pfctl
ALL ALL=NOPASSWD: /sbin/pfctl -s state
```

I'm running macOS Catalina 10.15.7.

```sh
$ mitmproxy --version
Mitmproxy: 7.0.4
Python:    3.9.9
OpenSSL:   OpenSSL 1.1.1l  24 Aug 2021
Platform:  macOS-10.15.7-x86_64-i386-64bit
```

I've given my phone a manual IP with `router` set to the IP of my Mac: 192.168.1.77 (`en0`, see below).

```sh
en0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	options=400<CHANNEL_IO>
	ether 38:f9:d3:9e:28:9f 
	inet6 fe80::be:2e35:bbd9:1db8%en0 prefixlen 64 secured scopeid 0x6 
	inet 192.168.1.77 netmask 0xffffff00 broadcast 192.168.1.255
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect
	status: active
```

But it won't connect using WiFi (as you can see on the screenshots I only get 4G).

What am I doing wrong? How can I troubleshoot further.

Thanks 🙏 

![IMG_6315](https://user-images.githubusercontent.com/15611634/145433480-9e228c8b-e127-4ef0-a1c5-27a419e987a4.PNG)
![IMG_6316](https://user-images.githubusercontent.com/15611634/145433484-c3890b55-db0a-4bb3-b527-004b2affbd00.PNG)

Ok, made (some) progress.

I tethered my iPhone to my mac using a USB cable and updated the `pf.conf` to use `bridge100` instead of `en0`.

However, now mitm crashes...

I'm very new to this field, but it looks like what's happening is that the TLS handshake fails, and when mitm tries to "force" it, the server disconnects which mitm can't handle and then pukes (`Unexpected EOF`).

But what's the root issue here, why does it reject the cert?

<img width="1440" alt="Screenshot 2021-12-09 at 21 46 03" src="https://user-images.githubusercontent.com/15611634/145473792-046c1d96-6ead-4e1d-ba20-a9d9fc6b49f0.png">
<img width="1440" alt="Screenshot 2021-12-09 at 21 46 57" src="https://user-images.githubusercontent.com/15611634/145473821-d4cd9e5b-851a-4b2b-a0be-4afa64593555.png">

I wonder if this is what I've run into: https://docs.mitmproxy.org/stable/concepts-certificates/#certificate-pinning

The app/company, [Ajax Systems](https://ajax.systems/), is extremely security aware, so it wouldn't surprise me.

**Comment by mhils** [comment]
Your current error seems to be that the server TLS handshake fails because mitmproxy does not trust the server's certificate. As a quick workaround you can set `ssl_insecure` to true and see if that gets you further.
